glide-core/src/client/value_conversion.rs

```diff
@@ -640,6 +687,256 @@ mod tests {
+                        Value::BulkString(b"field3".to_vec()),
+                        Value::BulkString(b"value3".to_vec()),
+                    ]),]),
+                ),
+            ]),
+            *value,
+        );
+    }
+
+    #[test]
+    fn test_convert_map_with_inner_array_to_map_of_maps_using_expected_return_type_map() {
+        // in RESP3, we get a map of arrays value like this:
+        // 1# "key1" =>
+        //    1) 1) "streamid-1"
+        //       2) 1) "f1"
+        //          2) "v1"
+        //          3) "f2"
+        //          4) "v2"
+        //    2) 1) "streamid-2" ...
+        // 2# "key2" => ...
+        //
+        let map_of_arrays = vec![
+            (
+                Value::BulkString("key1".into()),
+                Value::Array(vec![Value::Array(vec![
+                    Value::BulkString(b"streamid-1".to_vec()),
+                    Value::Array(vec![
+                        Value::BulkString(b"field1".to_vec()),
+                        Value::BulkString(b"value1".to_vec()),
+                    ]),
+                ])]),
+            ),
+            (
+                Value::BulkString("key2".into()),
+                Value::Array(vec![
+                    Value::Array(vec![
+                        Value::BulkString(b"streamid-2".to_vec()),
+                        Value::Array(vec![
+                            Value::BulkString(b"field21".to_vec()),
+                            Value::BulkString(b"value21".to_vec()),
+                            Value::BulkString(b"field22".to_vec()),
+                            Value::BulkString(b"value22".to_vec()),
+                        ]),
+                    ]),
+                    Value::Array(vec![
+                        Value::BulkString(b"streamid-3".to_vec()),
+                        Value::Array(vec![
+                            Value::BulkString(b"field3".to_vec()),
+                            Value::BulkString(b"value3".to_vec()),
+                        ]),
+                    ]),
+                ]),
+            ),
+        ];
+
+        // convert to a map value like this:
+        // #1) "key1"
+        //    #1) "streamid-1"
+        //         1) "f1"
+        //         2) "v1"
+        //         3) "f2"
+        //         4) "v2"
+        //    #2) "streamid-2"
+        //    ...
+        // #2) "key2"
+        // ...
+        let mut cmd = redis::cmd("XREAD");
+        let expected_type = expected_type_for_cmd(cmd.arg("STREAMS").arg("key").arg("id"));
+        let converted_map =
+            convert_to_expected_type(Value::Map(map_of_arrays), expected_type).unwrap();
+
+        let converted_map = if let Value::Map(map) = converted_map {
+            map
+        } else {
+            panic!("Expected a Map, but got {:?}", converted_map);
+        };
+
+        assert_eq!(converted_map.len(), 2);
+
+        let (key, value) = &converted_map[0];
+        assert_eq!(Value::BulkString(b"key1".to_vec()), *key);
```

Comment:
I think it's clearer what Yuri did in his PR - store the expected Value in a variable and assert that the converted_map is equals it